adds attributes for more fine-grained vertical and horizontal positio…

jwawi opened the pull request:
Adds attributes for additional fine-grained horizontal and vertical positioning of tourtip elements: tourtip-vertical-offset and tourtip-horizontal-offset. Retains original tourtip-offset attribute functionality for users of previous versions.

booleanbetrayal commented on src/tour/tour.js:
```diff
@@ -194,6 +194,14 @@ angular.module('angular-tour.tour', [])
           scope.ttOffset = parseInt(val, 10) || tourConfig.offset;
         });
 
+        attrs.$observe( 'tourtipVerticalOffset', function ( val ) {
```
Would you mind renaming the `verticalOffset` and `horizontalOffset` properties to `offsetVertical` and `offsetHorizontal` just so the properties' namings align a little better?

booleanbetrayal commented on src/tour/tour.js:
```diff
@@ -194,6 +194,14 @@ angular.module('angular-tour.tour', [])
           scope.ttOffset = parseInt(val, 10) || tourConfig.offset;
```
this should now set `scope.horizontalOffset` instead of `scope.ttOffset`

jwawi replied:
Wouldn't doing so break legacy functionality? scope.ttOffset was originally used to alter horizontal positioning on left and right aligned tourtips, but it alters vertical positioning on bottom, top, and center-top positioned tourtips in the calculatePosition() function.

booleanbetrayal replied:
yeah, you're right. I'm starting to re-evaluate the backwards compatibility seeing as how `scope.ttOffset` is used for both vertical and horizontal offsets now. Perhaps its best to just axe `tourOffset` / `scope.ttOffset` entirely and just introduce the breaking change, since it's cleaner / more easily understandable.

I'd be fine if you did that, and updated the tests / demo app accordingly.

jwawi replied:
If we're axing scope.ttOffset, should we make those available to configure globally in the tourConfig service then?

booleanbetrayal replied:
Only issue with axing `scope.ttOffset` is that it's used very specifically related to the positioning desired (left vs right), etc. Sorry for the back and forth on this, we may need to retain it as it's effectively acting as a default padding / margin. 

So here's my final suggestion: 

We retain the existing offset functionality, but rename it to "margin" instead, so as not to be confusing with the new offset properties. So the only changes required would be renaming things and clarifying the documentation; the calculations would remain as they are in your PR.

I don't think that we'd need to expose the vertical / horizontal offsets as global config values, because I can't see an immediate use case where all tips would need positioning adjustments ... but maybe i'm missing a use-case.

Sound good?

jwawi replied:
Sounds good. The only use case I could think of where you might want to globally define is if the walkthrough template is very dynamic and nothing is really on the page at compilation, but it's probably not necessary and would be something that a developer would need to specify on each element.

I have the documentation updated locally, updated src/tour/tour.js to use tourConfig.margin and scope.ttMargin, and ran a grunt build to update dist. The tests don't reference offset so they should be fine. I will commit my changes, rebase, and push them out.

booleanbetrayal commented:
Cool ... see the comments above. Would also be good to update the documentation!

jwawi commented:
Everything should be updated now. I rebased/squashed the iterative commits from earlier, updated the documentation, changed scope.ttOffset to scope.ttMargin, changed tourtip-offset to tourtip-margin, and updated the attribute names to tourtip-offset-vertical/scope.offsetVertical and tourtip-offset-horizontal/scope.offsetHorizontal.

I also ran a grunt build to update the dist files. The tests didn't reference offset so they should still be good I think. Take a look and let me know what you think.

booleanbetrayal commented:
sweet ... thanks for this PR @jwawi !

jwawi commented:
my pleasure! happy to contribute
